backend/src/main/java/co/inventorsoft/scripty/service/ProjectServiceInterfaceImpl.java

```diff
@@ -0,0 +1,36 @@
+package co.inventorsoft.scripty.service;
+
+import co.inventorsoft.scripty.exception.ApplicationException;
+import co.inventorsoft.scripty.model.dto.ProjectToFront;
+import co.inventorsoft.scripty.model.entity.User;
+import co.inventorsoft.scripty.repository.ProjectRepository;
+import co.inventorsoft.scripty.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+
+@Service
+public class ProjectServiceInterfaceImpl implements ProjectServiceInterface {
+
+    ProjectRepository projectRepository;
+    UserRepository userRepository;
+
+    @Autowired
+    public ProjectServiceInterfaceImpl(ProjectRepository projectRepository, UserRepository userRepository) {
+        this.projectRepository = projectRepository;
+        this.userRepository = userRepository;
+
+    }
+
+    public Collection<ProjectToFront> getProjectList(Authentication auth) {
+        if(auth==null) {
```

Comment:
It's very simplified (and wrong, strictly speaking) implementation. You'd better look for [Specification API](https://spring.io/blog/2011/04/26/advanced-spring-data-jpa-specifications-and-querydsl/). It's designed to support dynamic queries and it will be easier to extend if we use them